phys2bids/phys2bids.py

```diff
@@ -309,71 +215,160 @@ def phys2bids(filename, info=False, indir='.', outdir='.', heur_file=None,
-                # with the same frequency
-                phys_out[uniq_freq].delete_at_index(idx - count)
-                # take into acount the elimination so in the next eliminated channel we
-                # eliminate correctly
-                count += 1
-        # Also create a BlueprintOutput object for each unique frequency found.
-        # Populate it with the corresponding blueprint input and replace it
-        # in the dictionary.
-        phys_out[uniq_freq] = BlueprintOutput.init_from_blueprint(phys_out[uniq_freq])
+    # Checking acquisition type via user's input
+    if tr != [0, ] and num_timepoints_expected != [0, ]:
+
+        #  Multi-run acquisition type section
+        #  Check list length, more than 1 means multi-run
+        if len(num_timepoints_expected) > 1:
+            # if multi-run of same sequence type, pad list with ones
+            # and multiply array with user's input
+            if len(tr) == 1:
+                tr = ones(len(num_timepoints_expected)) * tr[0]
+            # Check equivalency of length
+            elif len(num_timepoints_expected) != len(tr):
+                raise Exception('Number of sequence types listed with TR '
+                                'doesn\'t match expected number of runs in '
+                                'the session')
+
+            # Sum of values in ntp_list should be equivalent to num_timepoints_found
+            phys_in.check_trigger_amount(thr=thr,
+                                         num_timepoints_expected=sum(num_timepoints_expected),
+                                         tr=1)
+
+            # Check that sum of tp expected is equivalent to num_timepoints_found,
+            # if it passes call slice4phys
+            if phys_in.num_timepoints_found != sum(num_timepoints_expected):
+                raise Exception('The number of triggers found is different '
+                                'than expected. Better stop now than breaking '
+                                'something.')
+
+            # slice the recording based on user's entries
+            # !!! ATTENTION: PHYS_IN GETS OVERWRITTEN AS DICTIONARY
+            phys_in = slice4phys(phys_in, num_timepoints_expected, tr, thr)
+            # returns a dictionary in the form {run_idx: phys_in[startpoint, endpoint]}
+
+            # save a figure for each run | give the right acquisition parameters for runs
+            for i, run in enumerate(phys_in.keys()):
+                viz.export_trigger_plot(phys_in[run], num_timepoints_expected[i], tr[i],
+                                        chtrig, outdir, filename, sub, ses, run)
+            # define run amount
+            run_amount = len(phys_in)
+
+        # Single run acquisition type, or : nothing to split workflow
+        else:
+            # Run analysis on trigger channel to get first timepoint
+            # and the time offset.
+            phys_in.check_trigger_amount(chtrig, thr, num_timepoints_expected[0], tr[0])
```

Comment:
```suggestion
            phys_in.check_trigger_amount(thr, num_timepoints_expected[0], tr[0])
```

`check_trigger_amount()` does not take `chtrig` as a parameter